jenkins-infra/plugin-health-scoring
# Dockerize the Application

Closes #4

This PR mainly includes (inside the location `/src/main/docker/`):
* `Dockerfile`: to create an image of our Spring Boot Application.
* `create_tables.sql`: to create both the required database and table.
* `docker-compose.yml`: main file where we define definition to containerize both the DB and Spring Boot Application.
* `application.properties`: where we replaced the previously used encryption mechanism using `jasypt` library with the environment variables.

Steps to locally setup the application:
1. `$ mvn clean package -DskipTests`
2. `$ cp target/*.jar src/main/docker/`
3. `$ docker-compose up`

Steps to test if the application works properly:
1. `$ docker exec -it <container name of the DB> psql <database name> -U <username> -c 'select * from plugins;'`
2. You should see a table with column names 'id', 'name', 'release_timestamp', and 'scm', whose rows are filled with plugin details.
